piksi_tools/console/fusion_engine_status.py

```diff
@@ -0,0 +1,153 @@
+from threading import Timer
+
+from sbp.system import SBP_MSG_INS_UPDATES, MsgInsUpdates
+from traits.api import Enum, HasTraits, List, Instance, Dict, Str
+from traitsui.api import (Item, View, HGroup)
+
+
+# No updates have been attempted in the past `STATUS_PERIOD`
+UNKNOWN = u'\u2B1B'  # Unicode Character “⬛” (U+2B1B)
+# There have been attempted updates in the past `STATUS_PERIOD` but at least one was rejected
+WARNING = u'\u26A0'  # Unicode Character “⚠” (U+26A0)
+# There have been updates in the past `STATUS_PERIOD` and none were rejected
+OK = u'\u26AB'  # Unicode Character “⚫” (U+26AB)
+
+
+# MsgInsUpdates flags
+GNSSPOS = 'gnsspos'
+GNSSVEL = 'gnssvel'
+WHEELTICKS = 'wheelticks'
+SPEED = 'speed'
+NHC = 'nhc'
+ZEROVEL = 'zerovel'
+
+ALL_FLAGS = [
+    GNSSPOS,
+    GNSSVEL,
+    WHEELTICKS,
+    SPEED,
+    NHC,
+    ZEROVEL,
+]
+
+FLAG_LABELS = {
+    GNSSPOS: 'GNSS Pos',
+    GNSSVEL: 'GNSS Vel',
+    WHEELTICKS: 'Wheelticks',
+    SPEED: 'Wheelspeed',
+    NHC: 'nhc',
+    ZEROVEL: 'Static Detection',
+}
+
+STATUS_PERIOD = 1
+
+
+def delayed(old, func, delay=STATUS_PERIOD):
+    if old is not None:
+        old.cancel()
+    t = Timer(delay, func)
+    t.start()
+    return t
+
+
+def status_item(status):
+    return Item(
+        'status',
+        show_label=False,
+        visible_when='status == \'{}\''.format(status),
+        springy=False,
+        style='readonly',
+        style_sheet=status_to_style(status)
+    )
+
+
+def status_to_style(status):
+    if status == OK:
+        return '* { color: green; }'
+    elif status == WARNING:
+        return '* { color: orange; }'
+    else:
+        return '* { color: grey; }'
+
+
+def check_flag(flag):
+    return lambda msg: (getattr(msg, flag) & 0b1111) == 0
+
+
+def stats_key(flag):
+    return f'stats_{flag}'
```

Comment:
We're using Python 3.5 for deploying currently 😬 -- so need to transform this to "old style" string interpolation